### PR TITLE
Set term actif in contract form - corrections from preview PR

### PR DESCRIPTION
**Pull request by jeremie74**

### CODE REVIEW

File: libs/admin/src/lib/admin-panel/pages/offers/shell/contract-form/contract-form.component.html [hunk 1]
```diff
@@ -15,12 +15,12 @@
         <mat-error *ngIf="form.hasError('required', 'price')">Price field is required!</mat-error>
       </mat-form-field>
     </header>
-    
-    <bf-form-table [form]="form.get('terms')" [columns]="termColumns" tablePosition="bottom">
+    {{indexId}}
```
Comment (GrandSchtroumpf) [inline libs/admin/src/lib/admin-panel/pages/offers/shell/contract-form/contract-form.component.html:18]: can you remove ?

File: libs/admin/src/lib/admin-panel/pages/offers/shell/contract-form/contract-form.component.ts [hunk 2]
```diff
@@ -40,7 +40,8 @@ export class ContractFormComponent implements OnInit {
   })
   titles$ = this.service.valueChanges(ref => ref.where('app.catalog.status', '==', 'approved'));
   currency?: string;
-  showBfFilterTable = false;
+  termId: string;
```
Comment (GrandSchtroumpf) [inline libs/admin/src/lib/admin-panel/pages/offers/shell/contract-form/contract-form.component.ts:43]: you don't need to store this value in the properties.

File: libs/admin/src/lib/admin-panel/pages/offers/shell/contract-form/contract-form.component.ts [hunk 3]
```diff
@@ -84,7 +77,15 @@ export class ContractFormComponent implements OnInit {
       this.currency = offer?.currency;
       const terms = await this.termService.getValue(contract.termIds);
       (this.form.get('terms') as FormList<any>).patchAllValue(terms);
-    }
+
+      // We show the term sent in the Query from the view terms page
+      if (this.route.snapshot.queryParams.termId) {
+        this.termId = this.route.snapshot.queryParams.termId;
```
Comment (GrandSchtroumpf) [inline libs/admin/src/lib/admin-panel/pages/offers/shell/contract-form/contract-form.component.ts:83]: don't store this value in this the property as you don't use it after. : 
```typescript 
const termId = this.route.snapshot.queryParams.termId;
if (termId) {
  const terms = this.form.get('terms').value;
  this.indexId = terms .findIndex(value => value.id === termId);
}
```

File: libs/ui/src/lib/form/table/form-table.component.ts [hunk 4]
```diff
@@ -84,10 +88,11 @@ export class FormTableComponent<T> implements OnInit, AfterViewInit, OnDestroy {
     // Keep the table updated
     this.sub = values$.subscribe(values => {
       this.dataSource.data = values;
-      this.cdr.markForCheck();
     });
 
     this.add();
+
+    this.cdr.markForCheck();
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/form/table/form-table.component.ts:95]: I'm pretty sure you don't need this line.

**Review by GrandSchtroumpf — CHANGES_REQUESTED** [review]

File: libs/admin/src/lib/admin-panel/pages/offers/shell/contract-form/contract-form.component.html (hunk 1 above)
Comment (jeremie74) [reply libs/admin/src/lib/admin-panel/pages/offers/shell/contract-form/contract-form.component.html:18]: Oops sorry

File: libs/ui/src/lib/form/table/form-table.component.ts [hunk 5]
```diff
@@ -84,7 +88,6 @@ export class FormTableComponent<T> implements OnInit, AfterViewInit, OnDestroy {
     // Keep the table updated
     this.sub = values$.subscribe(values => {
       this.dataSource.data = values;
-      this.cdr.markForCheck();
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/form/table/form-table.component.ts:87]: Sorry I did a mistake in my last comment,  you might need this one actually.

**Review by GrandSchtroumpf — APPROVED** [review]

**Pull request merged**